### DISCUSSION TITLE
swagger 사용 관련 의논

### DISCUSSION THREAD

**Discussion by kang20** [LikeLion12th-OMEB/Backend#32] [Q&A] [answered]
# 문제 1
![image](https://github.com/user-attachments/assets/2eb4fef1-c45d-4383-87b0-ccd079b0d463)
위처럼 Swagger를 사용하면 메서드 위에 어노테이션을 많이 적어야함
가독성도 뭔가 떨어지고 지저분한 느낌이 들음

### 대안 1
다른 사람 레포를 봤을 때 Controller 인터페이스를 만들고 인터페이스에 swagger 관련 어노테이션을 몰아 넣었음
대충 ReviewController 가 있으면
ReviewControllerApi 로 만들어서 거기에 swagger 관련 어노테이션을 명시한 곳이 있어서 좋아보여서 도입하면 어떨까 의견 나누고자 함

### 대안 2
그냥 이대로 진행

# 문제 2
![image](https://github.com/user-attachments/assets/b709b5b0-7e07-41f7-ac3d-35af55cc2184)
위 사진처럼 controller 인자에 @Schema(hidden = true) 를  해줘야 swagger에 userId가 안나타남

## 대안 1
원래 @UserPrincipal 어노테이션으로 principal을 들고 오는데 우린 AOP로 적용해서 컨트롤러 위에 어노테이션을 적용함
@UserPrincipal 와  @Schema(hidden = true) 를 상속 받은 어노테이션을 커스텀해서 적용하면 어떨까 함

## 대안 2
그냥 이대로 진행

**Comment by siwan9** [comment, marked as answer]
두 문제 모두 대안1이 좋아보이네요
